app/Contracts/Events/HasTable.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+/**
+ * SPDX-License-Identifier: MIT
+ * Copyright (c) 2017-2018 Tobias Reich
+ * Copyright (c) 2018-2025 LycheeOrg.
+ */
+
+namespace App\Contracts\Events;
+
+interface HasTable
+{
```

Comment:
why not an abstract class that has both `public function metricAction(): MetricsAction;` & `public function table(): string;` and called `MetricEvent` for example? si there going to be an a MetricAction without storign it anywhere?